mipengine/node/monetdb_interface/common_actions.py

```diff
@@ -13,15 +13,25 @@
 
 
 def create_table_name(
-    table_type: TableType, command_id: str, context_id: str, node_id: str
+    table_type: TableType,
+    node_id: str,
+    context_id: str,
+    command_id: str,
+    subcommand_id: str = None,
 ) -> str:
     """
     Creates and returns in lower case a table name with the format <tableType>_<commandId>_<contextId>_<nodeId>
+
+    A suffix to command_id could be added, if a subcommand_id is provided.
+    The subcommand_id is used when there are multiple table definitions on the same command.
     """
     if table_type not in {TableType.NORMAL, TableType.VIEW, TableType.MERGE}:
         raise TypeError(f"Table type is not acceptable: {table_type} .")
 
-    return f"{table_type}_{command_id}_{context_id}_{node_id}".lower()
+    if subcommand_id:
+        command_id = f"{command_id}_{subcommand_id}"
+
+    return f"{table_type}_{node_id}_{context_id}_{command_id}".lower()
```

Comment:
this changes the table naming convention substantially. The problem with sometimes returning this 
```
<table_type_nodeid_contextid_commandid>
```
and sometimes this
```
<table_type_nodeid_contextid_commandid_subcommandid>
```
is that if there is a consumer that needs to do something with these values it will have to split and then check the number of fields. Since it seems there is actually a need for another field (a commandsubid) let's always put it there so there is a single convention instead of two
```
     return f"{table_type}_{node_id}_{context_id}_{command_id}_{command_subid}".lower() 
```

Reply:
When a "command_subid" is not provided what should the value be?

Reply:
Should the value be "0" ?

Reply:
yes, in ./mipengine/node/tasks/udfs.py l.185 it is anyway assigned as the index of the udf_execution_queries.output_tables so no big change there, but it seems it will need a lot of small changes in other places..

Reply:
Fixed